### PR TITLE
Add property-based testing to DateTime serialisation

### PR DESCRIPTION
**Pull request by bigmontz**

### CODE REVIEW

File: packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js [hunk 1]
```diff
@@ -850,6 +852,84 @@ describe('#unit BoltProtocolV4x3', () => {
...
+                unpacked.day,
+                unpacked.hour,
+                unpacked.minute,
+                unpacked.second,
+                unpacked.nanosecond,
+                undefined,
+                unpacked.timeZoneId
+              )
+
+              expect(loggerFunction)
```
Comment (robsdedude) [inline packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js]: Is this expected to happen during packing or unpacking? As the test is right now, either would be accepted.

File: packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js [hunk 2]
```diff
@@ -850,6 +852,84 @@ describe('#unit BoltProtocolV4x3', () => {
...
       })
 
+      it('should pack and unpack DateTimeWithOffset', () => {
+        fc.assert(
+          fc.property(
+            fc.date({
+              min: temporalUtil.newDate(utils.MIN_UTC_IN_MS + utils.ONE_DAY_IN_MS),
+              max: temporalUtil.newDate(utils.MAX_UTC_IN_MS - utils.ONE_DAY_IN_MS)
+            }),
+            utils.arbitraryTimeZoneId(),
```
Comment (robsdedude) [inline packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js:862]: Consider adding another `fc.integer({min: 0, max: 999_999})` and add it on top of `date.getUTCMilliseconds() * 1_000_000`. This way, you cover the full ns range.

File: packages/bolt-connection/test/bolt/bolt-protocol-v5x0.test.js [hunk 3]
```diff
@@ -587,6 +589,102 @@ describe('#unit BoltProtocolV5x0', () => {
...
+            const loggerFunction = jest.fn()
+            const protocol = new BoltProtocolV5x0(
+              new utils.MessageRecordingConnection(),
+              buffer,
+              {
+                disableLosslessIntegers: true
+              },
+              undefined,
+              new Logger('debug', loggerFunction)
+            )
```
Comment (robsdedude) [inline packages/bolt-connection/test/bolt/bolt-protocol-v5x0.test.js:623]: Why is this not in a beforeEach hook for 5.0 like for the other protocol versions?

**Review by robsdedude — APPROVED** [review]
:test_tube: :calendar:

File: packages/bolt-connection/test/bolt/bolt-protocol-v5x0.test.js (hunk 3 above)
Comment (bigmontz) [reply packages/bolt-connection/test/bolt/bolt-protocol-v5x0.test.js:623]: We don't have a subset of test for this scenario in the v5

File: packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js (hunk 1 above)
Comment (bigmontz) [reply packages/bolt-connection/test/bolt/bolt-protocol-v4x3.test.js]: During the packing, I will move the expectation to just after the packing.

**Pull request merged**